tests/run-make/deref-impl-rustdoc-ice/rmake.rs

```diff
@@ -0,0 +1,16 @@
+// A very specific set of circumstances (mainly, implementing Deref, and
+// having a procedural macro and a Debug derivation in external crates) caused
+// an internal compiler error (ICE) when trying to use rustdoc. This test
+// reproduces the exact circumstances which caused the bug and checks
+// that it does not happen again.
```

Comment:
Not-a-review-comment: this is *very* specific... it must've been a pain to figure out lol

Reply:
I know, right? I find it especially hilarious that the fix for all this, in the linked PR, is literally just adding a single line of code.